pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/RegexpLikePredicateEvaluatorFactory.java

```diff
@@ -64,6 +65,7 @@ public static BaseRawValueBasedPredicateEvaluator newRawValueBasedEvaluator(Rege
   private static final int PATTERN_FLAG = Pattern.UNICODE_CASE | Pattern.CASE_INSENSITIVE;
 
   private static final class DictionaryBasedRegexpLikePredicateEvaluator extends BaseDictionaryBasedPredicateEvaluator {
+    private final Matcher _matcher;
```

Comment:
(minor) Do not add `private` in a `private` inner class

Reply:
OK, I will move the compilation to `RegexpLikePredicate` too, in order to deduplicate some code.